packages/shared/src/hooks/useInputFieldFunctions.ts

```diff
@@ -79,10 +79,10 @@ function useInputFieldFunctions<
   }, [validInput]);
 
   useEffect(() => {
-    if (validInput !== undefined && valid !== undefined) {
```

Comment:
I am afraid that this was here for a reason. To avoid the re-renders, I saw from the history that we've had this removed from the dependency list.

One way to work it around, as I don't want to disable the lint when removing a dependency - we can introduce a `useRef` that will hold the data of `validInput`. Then a `useEffect` to update the ref so whenever this updated `useEffect` would still have the same checks.